### PR TITLE
Fix/threading

### PR DESCRIPTION
**Pull request by aclement455**

### CODE REVIEW

File: android/src/main/java/com/guichaguri/trackplayer/service/Utils.java [hunk 1]
```diff
@@ -10,6 +10,7 @@
 import android.net.Uri;
 import android.os.Build;
 import android.os.Bundle;
+import android.util.Log;
 import android.support.v4.app.NotificationCompat;
 import android.support.v4.content.LocalBroadcastManager;
```
Comment (pawelnuzka) [inline android/src/main/java/com/guichaguri/trackplayer/service/Utils.java]: Unused import

File: android/src/main/java/com/guichaguri/trackplayer/service/Utils.java [hunk 2]
```diff
@@ -191,11 +192,15 @@ public static Notification createBlankSetupNotification(Context context) {
     }
 
     public static void emit(Context context, String event, Bundle data) {
-        Intent intent = new Intent(Utils.EVENT_INTENT);
```
Comment (pawelnuzka) [inline android/src/main/java/com/guichaguri/trackplayer/service/Utils.java:194]: `EVENT_INTENT` not used anymore same as `PLAYBACK_SERVICE_SETUP_LOCK`

File: android/src/main/java/com/guichaguri/trackplayer/service/MusicService.java [hunk 3]
```diff
@@ -2,44 +2,27 @@
...
-import com.facebook.react.bridge.Arguments;
-import com.facebook.react.bridge.ReactContext;
-import com.facebook.react.jstasks.HeadlessJsTaskConfig;
-import javax.annotation.Nullable;
+import com.guichaguri.trackplayer.service.metadata.MetadataManager;
 
 /**
  * @author Guichaguri
  */
-public class MusicService extends HeadlessJsTaskService {
```
Comment (pawelnuzka) [inline android/src/main/java/com/guichaguri/trackplayer/service/MusicService.java:27]: Unused imports:
`import android.app.Activity;`
`import android.content.Context;`
 `import com.facebook.react.HeadlessJsTaskService;`

File: android/src/main/java/com/guichaguri/trackplayer/module/MusicEvents.java [hunk 4]
```diff
@@ -1,5 +1,6 @@
 package com.guichaguri.trackplayer.module;
 
+import android.util.Log;
```
Comment (pawelnuzka) [inline android/src/main/java/com/guichaguri/trackplayer/module/MusicEvents.java]: All imports in `MusicEvents.java` are unused.

File: android/src/main/java/com/guichaguri/trackplayer/module/MusicModule.java [hunk 5]
```diff
@@ -49,21 +51,6 @@ public String getName() {
     public void initialize() {
```
Comment (pawelnuzka) [inline android/src/main/java/com/guichaguri/trackplayer/module/MusicModule.java]: No need for `initialize` method at all

File: android/src/main/java/com/guichaguri/trackplayer/module/MusicModule.java [hunk 6]
```diff
@@ -49,21 +51,6 @@ public String getName() {
...
         ReactContext context = getReactApplicationContext();
         LocalBroadcastManager manager = LocalBroadcastManager.getInstance(context);
-
-        eventHandler = new MusicEvents(context);
-        manager.registerReceiver(eventHandler, new IntentFilter(Utils.EVENT_INTENT));
-    }
-
-    @Override
-    public void onCatalystInstanceDestroy() {
```
Comment (pawelnuzka) [inline android/src/main/java/com/guichaguri/trackplayer/module/MusicModule.java:58]: Are you sure we should not unbind the service here, eg:
```
if(binder != null || connecting) {
   getReactApplicationContext().unbindService(this);
}
```

File: android/src/main/java/com/guichaguri/trackplayer/service/MusicService.java [hunk 7]
```diff
@@ -56,86 +39,43 @@ public void destroy() {
...
     @Override
     public void onDestroy() {
         super.onDestroy();
-
         destroy();
     }
-
-    @Override
-    public void onTaskRemoved(Intent rootIntent) {
-        super.onTaskRemoved(rootIntent);
```
Comment (pawelnuzka) [inline android/src/main/java/com/guichaguri/trackplayer/service/MusicService.java]: Is this truly not needed anymore? Where do we stop the service?

**Review by pawelnuzka — APPROVED** [review]
Good work, some minor comments about imports and stopping the service

Tested on Samsug Galaxy J9 and the performance is better, moving `MusicService` to separate thread and decreasing intervals of get state/position calls seemed to help. Tested on background and all features work, didn't get any crashes.

**Comment by biomancer** [comment]
Hey guys! Did these changes work out for you? Maybe you could create a pr to main repo with some basic explanation about idea behind it?

**Pull request merged**